test/lib/Benchmark.js

```diff
@@ -0,0 +1,182 @@
+/* globals XPCOMUtils, Services, Task */
+"use strict";
+
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "Task",
+                                  "resource://gre/modules/Task.jsm");
+// For the performance.now()
+const performance = Services.appShell.hiddenDOMWindow.performance;
+
+/*
+ * A simple benchmark framework that runs on top of jpm test
+ *
+ * Note that the timing resolution for the benchmark is microsecond
+ *
+ * Example:
+
+  function fib(n) {
+    if (n < 2) {
+      return n;
+    }
+
+    return fib(n - 1) + fib(n - 2);
+  }
+
+  exports.test_benchmark = function*() {
+    let bench = new Benchmark(b => {
+      for (let i = 0; i < b.N; i++) {
+        fib(20);
+      }
+    });
+    let result = yield bench.launch();
+    console.log(result);
+  };
+*/
+
+/*
+ * {Params} func is the benchmark function/generator that is of the following form
+ *
+   function(b) {
+     for (let i = 0; i < b.N; i++) {
+       // do your stuff here
+     }
+   }
```

Comment:
Out of curiousity, why does the API make the caller do the looping?  It's probably worth at least a comment...